File: Solutions/1492-KthFactorOfN/KthFactorOfN.java
class Solution {
    public int kthFactor(int n, int k) {
        int curr = 1;
        while (curr <= n){
          if(n % curr == 0){
            k--;
            if(k == 0){
              return curr;
             }
            }
            curr++;
        }
        
        return -1;
    }
}

File: Solutions/1492-KthFactorOfN/KthFactorOfN.kt
class KthFactorOfNSolution {
    fun kthFactor(n: Int, k: Int): Int {
        var currFactor = 1
        var kthFactor = k

        while (currFactor <= n) {
            if (n % currFactor == 0) kthFactor--
            if (kthFactor == 0) return currFactor
            currFactor++
        }
        return -1
    }
}
